src/Synapse/Synapse/Commands/ManagementCommands/SqlPool/RemoveAzureSynapseSqlPoolRestorePoint.cs

```diff
@@ -0,0 +1,114 @@
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = HelpMessages.PassThru)]
+        public SwitchParameter PassThru { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = HelpMessages.AsJob)]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (this.IsParameterBound(c => c.SqlPoolObject))
+            {
+                var resourceIdentifier = new ResourceIdentifier(this.SqlPoolObject.Id);
+                this.ResourceGroupName = resourceIdentifier.ResourceGroupName;
+                this.WorkspaceName = resourceIdentifier.ParentResource;
+                this.WorkspaceName = this.WorkspaceName.Substring(this.WorkspaceName.LastIndexOf('/') + 1);
+                this.SqlPoolName = resourceIdentifier.ResourceName;
+            } 
+
+            if (this.IsParameterBound(c => c.InputObject))
+            {
+                this.Name = this.InputObject.Id.Substring(this.InputObject.Id.LastIndexOf('/') + 1);
+                this.ResourceId = this.InputObject.Id.Substring(0, this.InputObject.Id.Substring(0, this.InputObject.Id.LastIndexOf("/")).LastIndexOf("/"));
+                var resourceIdentifier = new ResourceIdentifier(this.ResourceId);
+                this.ResourceGroupName = resourceIdentifier.ResourceGroupName;
+                this.WorkspaceName = resourceIdentifier.ParentResource;
+                this.WorkspaceName = this.WorkspaceName.Substring(this.WorkspaceName.LastIndexOf('/') + 1);
+                this.SqlPoolName = resourceIdentifier.ResourceName;
+            }
+
+            if (this.IsParameterBound(c => c.ResourceId))
+            {
+                this.Name = this.ResourceId.Substring(this.ResourceId.LastIndexOf('/') + 1);
+                this.ResourceId = this.ResourceId.Substring(0,this.ResourceId.Substring(0, this.ResourceId.LastIndexOf("/")).LastIndexOf("/"));
+                var resourceIdentifier = new ResourceIdentifier(this.ResourceId);
+                this.ResourceGroupName = resourceIdentifier.ResourceGroupName;
+                this.WorkspaceName = resourceIdentifier.ParentResource;
+                this.WorkspaceName = this.WorkspaceName.Substring(this.WorkspaceName.LastIndexOf('/') + 1);
+                this.SqlPoolName = resourceIdentifier.ResourceName;
+            }
+
+            if (string.IsNullOrEmpty(this.ResourceGroupName))
+            {
+                this.ResourceGroupName = this.SynapseAnalyticsClient.GetResourceGroupByWorkspaceName(this.WorkspaceName);
+            }
+
+            if (this.ShouldProcess(this.Name, string.Format("Resources.RemovingSynapseSqlPoolRestorePoint", this.Name, this.ResourceGroupName, this.WorkspaceName, this.SqlPoolName)))
+            {
+                this.SynapseAnalyticsClient.DeleteSqlPoolRestorePoint(this.ResourceGroupName, this.WorkspaceName, this.SqlPoolName, this.Name);
```

Comment:
>DeleteSqlPoolRestorePoint [](start = 44, length = 25)

Do we need extra confirmation for this command?

Reply:
I agree!. I will look into the PR you post previously.